---
title: "Compiler Warning (level 1) CS1058 | Microsoft Docs"
ms.custom: ""
ms.date: "11/16/2016"
ms.prod: "visual-studio-dev14"
ms.reviewer: ""
ms.suite: ""
ms.technology: 
  - "devlang-csharp"
ms.tgt_pltfrm: ""
ms.topic: "error-reference"
f1_keywords: 
  - "CS1058"
dev_langs: 
  - "CSharp"
helpviewer_keywords: 
  - "CS1058"
ms.assetid: ed50590c-f130-47c3-976d-322a6c8f996d
caps.latest.revision: 11
caps.handback.revision: 11
author: "BillWagner"
ms.author: "wiwagn"
manager: "wpickett"
translationtype: Human Translation
---
# Compiler Warning (level 1) CS1058
Une clause catch précédente intercepte déjà toutes les exceptions. Toutes les exceptions levées seront encapsulées dans un System.Runtime.CompilerServices.RuntimeWrappedException  
  
 Cet attribut provoque l'avertissement CS1058 si un bloc `catch()` n'a aucun type d'exception spécifié après un bloc `catch (System.Exception e)`.  L'avertissement recommande que le bloc `catch()` n'intercepte pas les exceptions.  
  
 Un bloc `catch()` après un bloc `catch (System.Exception e)` peut intercepter des exceptions non conformes à CLS si `RuntimeCompatibilityAttribute` a la valeur false dans le fichier AssemblyInfo.cs : `[assembly: RuntimeCompatibilityAttribute(WrapNonExceptionThrows = false)]`.  Si cet attribut n'est pas défini explicitement avec la valeur false, toutes les exceptions levées non conformes à CLS sont encapsulées comme Exceptions et le bloc `catch (System.Exception e)` les intercepte.  Pour plus d'informations, consultez [Comment : intercepter une exception non\-CLS](../../../csharp/programming-guide/exceptions/how-to-catch-a-non-cls-exception.md).  
  
## Exemple  
 L'exemple suivant génère l'erreur CS1058.  
  
```  
// CS1058.cs  
// CS1058 expected  
using System.Runtime.CompilerServices;  
  
// the following attribute is set to true by default in the C# compiler  
// set to false in your source code to resolve CS1058  
[assembly: RuntimeCompatibilityAttribute(WrapNonExceptionThrows = true)]  
  
class TestClass   
{  
   static void Main()   
   {  
      try {}  
  
      catch (System.Exception e) {   
         System. Console.WriteLine("Caught exception {0}", e);  
      }  
  
      catch {}   // CS1058. This line will never be reached.  
   }  
}  
```